interfaces/seccomp/backend.go

```diff
@@ -162,6 +162,79 @@ func bpfBinPath(srcName string) string {
 	return filepath.Join(dirs.SnapSeccompDir, strings.TrimSuffix(srcName, ".src")+".bin")
 }
 
+func parallelCompile(compiler Compiler, profiles []string) error {
+	if len(profiles) == 0 {
+		// no profiles, nothing to do
+		return nil
+	}
+
+	profilesQueue := make(chan string, len(profiles))
+	N := runtime.NumCPU()
+	if N >= 2 {
+		N -= 1
+	}
+	if N > len(profiles) {
+		N = len(profiles)
+	}
+	M := N * 2
+	if M > len(profiles) {
+		M = len(profiles)
+	}
+	res := make(chan error, M)
+
+	// launch as many workers as we have CPUs
+	for i := 0; i < N; i++ {
+		go func() {
+			for {
+				profile, ok := <-profilesQueue
+				if !ok {
+					break
+				}
+				in := bpfSrcPath(profile)
+				out := bpfBinPath(profile)
+
```

Comment:
The original implementation removes existing profile before compiling, that's probably not critical?

Reply:
Good catch. Extended unit tests to cover this too.

Reply:
I think this may have some significance.

There's a bit of code in snap-confine that waits for up to 120 seconds until profiles are generated. This bit ensures that along with apparmor (which is updated instantly) seccomp effectively synchronises application startup to use the new profiles.

Reply:
Hmm, in that case perhaps we should have a comment in the code.

Reply:
@bboozzoo follow-up material.

Reply:
Since the build queue is stuck anyway, I've force pushed the comment update.